go/api/base_client.go

```diff
@@ -353,7 +361,19 @@ func (client *baseClient) executeCommandWithRoute(
 	)
 	client.mu.Unlock()
 
-	payload := <-resultChannel
+	// Wait for result or context cancellation
+	var payload payload
+	select {
+	case <-ctx.Done():
+		client.mu.Lock()
+		if client.pending != nil {
+			delete(client.pending, resultChannelPtr)
+		}
+		client.mu.Unlock()
+		return nil, ctx.Err()
```

Comment:
If GLIDE core returns a response after we stopped waiting for it, we will have a memory leak.
We probably need to start a goroutine there, and should wait for paiload to free the memory.

Reply:
We can do that in a follow up PR

Reply:
Will handle in separate PR. Issue logged on Jira board to capture comment.